proposals/07-datastore-optimization.md

```diff
@@ -0,0 +1,101 @@
+# Hi-speed Chain Store
+
+Authors: @hannahhoward @acruikshank @willscott @gammazero @mvdan
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/7
+
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+
+Chain syncing in Filecoin is painfully slow. This affects the whole product -- system requirements to run even a client node are very high, and gas prices and message limits are limited by the speed we can sync the chain. One huge source of this time suck is simply reading data in and out of our datastore.
+
+We have approached data persistence through using off the shelf key-value stores in our stack. Long term, many aspects of our stack are limited by the lack of a datastore/database designed specifically for our data format, IPLD (see this [overview](https://docs.google.com/document/d/1GxUNUaBfaY4HneWCOfyPcyWG_14t-vnsy3NbQRF4Fdc/edit) and [analysis](https://docs.google.com/document/d/16_8TEslIY__54x2lvMNu5dLL0BSAJ2GxkZ9w3p8sUys/edit)). But in the near term, we have several extremely promising alternative key/value store solutions that can deliver immediate concrete improvements to chain performance.
+
+The aim here is to finalize existing explorations to deliver at least a 50% improvement to chain performance.
+
+#### Assumptions &amp; hypotheses
```

Comment:
These are great. Agree there are risks here in this project’s scope and reliability at scale (would love comments from @magik6k @raulk and @anorth on these risks)